### DISCUSSION TITLE
Qt 5.7.1 msvc2019 static build

### DISCUSSION THREAD

**Discussion by oddMLan** [gonetz/GLideN64#2466] [General]
I reinstalled windows recently so I'm setting up my dev environment again. I can't find a link to the msvc2019 build, I thought we had one on the wiki. The revisions show the link did exist but it's broken. Precompiled builds don't go that back from what I see and they are dynamic builds. I found a newer static build on https://github.com/martinrotter/qt-minimalistic-builds but it doesn't go far back to 5.7.1. I tried to compile with a newer build and it fails with some linking errors. I hear GLideN64 links against the newest qt version pretty well on linux. Btw, are there any plans to upgrade the Qt version used? There are some benefits like proper DPI scaling of the UI. The current qt version used does not play well with high DPI displays. WTL works fine on HiDPI.

So if anyone can upload the specific version of qt required by GLideN64 that is compatible with msvc2019, I would appreciate it. Last time I tried compiling Qt from source it took 4 hours to compile on my machine and that makes my computer unusable during that time, so I would like avoiding that.

**Comment by gonetz** [comment]
I can't help with it. I use visual studio 2015 and 2017.
Regarding Qt version upgrade: I tried to do it a year ago. I took recent update but failed to build static version from sources. Newer version needs different set of flags and options, and I did not find a valid combination.

**Comment by Rosalie241** [comment]
> I can't find a link to the msvc2019 build

msvc2017 builds work fine with msvc2019.

> I tried to compile with a newer build and it fails with some linking errors

the qt libs changed filenames during the version changes, so to fix that you have to modify the static link lib list in the visual studio project files, i.e cmake detects those filenames 'dynamically' (each qt version ships with a cmake file which specifies everything) hence why it's 'easier' on linux.
